api/room/schema.py

```diff
@@ -79,8 +81,23 @@ def mutate(self, info, office_id, **kwargs):
         # remove block ID from kwargs, can't be saved in rooms model
         if kwargs.get('block_id'):
             kwargs.pop('block_id')
-        room = RoomModel(**kwargs)
+        room = RoomModel(
+            name=kwargs['name'],
+            room_type=kwargs['room_type'],
+            capacity=kwargs['capacity'],
+            image_url=kwargs['image_url'],
+            floor_id=kwargs['floor_id'],
+            calendar_id=kwargs['calendar_id']
+            )
         room.save()
+        for tag in kwargs['room_tags']:
+            room_tag = TagModel.query.filter_by(id=tag).first()
+            if not room_tag:
+                room.delete()
```

Comment:
You'll probably have to keep a lookout in case Michael's soft_delete PR is merged when it comes to the .delete() method